node/src/protocol/mod.rs

```diff
@@ -277,11 +281,18 @@ impl MpcSignProtocol {
                 continue;
             }
 
+            let after_step = SystemTime::now();
+
             let mut guard = self.state.write().await;
             *guard = state;
             drop(guard);
 
-            tokio::time::sleep(Duration::from_millis(1000)).await;
+            let step_duration = after_step.duration_since(before_step).unwrap().as_millis();
+
+            // If there was no work to do - sleep to avoid CPU and RPC burn
+            if step_duration < 2 {
+                tokio::time::sleep(Duration::from_millis(100)).await;
+            }
```

Comment:
we need similar logic for the error cases as well when they're sleeping for 1s

Reply:
I've set it to 100ms